### ISSUE TITLE
Provide better diagnostics when an assembly reference fails to load

### ISSUE THREAD

**Issue by DustinCampbell** [OmniSharp/omnisharp-roslyn#956]
There are a number of reasons that the MetadataFileReferenceCache might fail to load metadata from a file path. These need to be guarded against, and logged on failure. Otherwise, it can be difficult to diagnose when a project fails to get all of it's references.

Here's an example repro of when this might happen:

1. Have Windows on Linux (WSL) setup.
2. On Windows desktop, run `dotnet new console` to create a new project.
3. In WSL, run `dotnet restore` and `dotnet run`.
4. On Windows desktop, launch VS Code and load project.

The project will be filled with squiggles due to assembly references not being loaded at design-time because they're locked by the Unix subsystem.

**Comment by filipw** [comment]
Yeah, another example of the same thing is sharing a folder between mac OS and a Windows VM. I keep running into it all the time.

**Comment by DustinCampbell** [comment]
Yeah, we need to figure out why it's happening. Maybe it can be fixed. However, even if it can't, we shouldn't barf. :smile:
